test/test_backend/routes/tst_graph.py

```diff
@@ -32,70 +38,100 @@
-            print()
-
-
+        self.setUpClass()  # temp until tests finished
+        for test_type, test_name, codeset_ids, timeout_secs in [x.values() for x in self.test_cases]:
+            # TODO: temp
+            if test_name not in ['antibiotics 1']:
+                continue
+            input_path = THIS_INPUT_DIR / CONNECTED_SUBGRAPH_FILENAME_PATTERN.format(test_name)
+            sg, nodes_in_graph, preferred_concept_ids, orphans_not_in_graph, hidden = \
+                get_connected_subgraph(REL_GRAPH, codeset_ids, EXTRA_CONCEPT_IDS, HIDE_VOCABS)
+            #
+            normalized_hidden = {k: sorted(list(v)) for k, v in hidden.items()}
+            results = {
+                'sg.edges': sorted(list(sg.edges)),
+                'nodes_in_graph': sorted(list(nodes_in_graph)),
+                'preferred_concept_ids': sorted(list(preferred_concept_ids)),
+                'orphans_not_in_graph': sorted(list(orphans_not_in_graph)),
+                'hidden': normalized_hidden,
+            }
+            with open(input_path, 'w') as f:
+                f.write(json.dumps(results, indent=4))
+
+    async def tst_get_connected_subgraph(self, only_fast_cases=True):
+        """Test indented_concept_list()"""
+        self.setUpClass()  # temp until tests finished
+        for test_type, test_name, codeset_ids, timeout_secs in [x.values() for x in self.test_cases]:
+            # Do just 1 fast test case, or all test cases?
+            if only_fast_cases and test_name not in ['many-small', 'single-small', 'single-small-second-time']:
+                continue
+            # Expected
+            input_path = THIS_INPUT_DIR / CONNECTED_SUBGRAPH_FILENAME_PATTERN.format(test_name)
+            expected_raw = json.load(open(input_path, 'r'))
+            expected = {}  # convert JSON serialization back to sets
+            for k, v in expected_raw.items():
+                if k == 'sg.edges':
+                    expected[k] = {tuple(x) for x in v}
+                elif k == 'hidden':
+                    expected[k] = {k2: set(v2) for k2, v2 in v.items()}
+                else:
+                    expected[k] = set(v)
+            # Actual
+            sg, nodes_in_graph, preferred_concept_ids, orphans_not_in_graph, hidden = \
+                get_connected_subgraph(REL_GRAPH, codeset_ids, EXTRA_CONCEPT_IDS, HIDE_VOCABS)
+            actual = {
+                'sg.edges': set(sg.edges),
+                'nodes_in_graph': nodes_in_graph,
+                'preferred_concept_ids': preferred_concept_ids,
+                'orphans_not_in_graph': orphans_not_in_graph,
+                'hidden': hidden,
+            }
+            # Compare
+            for k, v in actual.items():
+                # self.assertEqual(v, expected[k], msg=f'Results differ for {k} for test {test_name}')
+                assert v == expected[k], f'Results differ for "{k}" in test for "{test_name}"'
```

Comment:
## Print set difference?
If you want, I could also print what was different when tests fail.

Reply:
siggie says not need